doc/list.rst

```diff
@@ -2465,6 +2465,15 @@ List of Available Rules
    Part of rule set `@PhpCsFixer:risky <./ruleSets/PhpCsFixerRisky.rst>`_
 
    `Source PhpCsFixer\\Fixer\\PhpUnit\\PhpUnitDataProviderNameFixer <./../src/Fixer/PhpUnit/PhpUnitDataProviderNameFixer.php>`_
+-  `php_unit_data_provider_return_type <./rules/php_unit/php_unit_data_provider_return_type.rst>`_
+
+   The type returned by data provider must be ``iterable``.
+
+   *warning risky* Risky when relying on signature of the data provider.
+
+   Part of rule set `@PhpCsFixer:risky <./ruleSets/PhpCsFixerRisky.rst>`_
```

Comment:
I believe we can now remove `testThatTestDataProvidersReturnIterableOrArray` method 🎉

Reply:
Done